privacyIDEA-impl/src/main/java/org/privacyidea/action/AbstractChallengeResponseAction.java

```diff
@@ -141,6 +146,24 @@ protected void extractChallengeData(@Nonnull PIResponse piResponse)
         {
             piFormContext.setPushMessage(piResponse.pushMessage());
         }
+
+        // Check for the images
+        List<Challenge> multiChallenge = piResponse.multichallenge;

```

Comment:
this line is pointless, use `piResponse.multichallenge` in the for-loop directly

Reply:
fixed [eab60eb](https://github.com/privacyidea/shibboleth-plugin/pull/38/commits/eab60eb4ea42a6d13dec507e8d9bc09eb43a9d27)